KsApi/Service.swift

```diff
@@ -536,4 +536,29 @@ public struct Service: ServiceType {
         )
         .flatMap(decodeModels)
   }
+
+  private func request<M: Decodable>(_ route: Route)
+    -> SignalProducer<M?, ErrorEnvelope> where M == M.DecodedType {
```

Comment:
@justinswart yes, the response is actually entirely empty for a `MessageThreadEnvelope` that doesn't contain messages. The other option was to modify every field in the `MessageThreadEnvelope` to be Optional, but we decided to add the `request` overload instead to avoid modifying the model.